tests/unit/s2n_client_key_share_extension_test.c

```diff
@@ -115,20 +118,54 @@ int main(int argc, char **argv)
             uint16_t actual_key_shares_size = s2n_stuffer_data_available(&key_share_extension);
             EXPECT_EQUAL(key_shares_size, actual_key_shares_size);
 
+            EXPECT_NOT_NULL(conn->config);
+            const struct s2n_ecc_preferences *ecc_preferences = conn->config->ecc_preferences;
+            EXPECT_NOT_NULL(ecc_preferences);
+
             /* should contain every supported curve, in order, with their sizes */
-            for (int i = 0; i < s2n_ecc_evp_supported_curves_list_len; i++) {
+            for (int i = 0; i < ecc_preferences->count; i++) {
                 uint16_t iana_value, share_size;
                 EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &iana_value));
-                EXPECT_EQUAL(iana_value, s2n_ecc_evp_supported_curves_list[i]->iana_id);
+                EXPECT_EQUAL(iana_value, ecc_preferences->ecc_curves[i]->iana_id);
                 EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &share_size));
-                EXPECT_EQUAL(share_size, s2n_ecc_evp_supported_curves_list[i]->share_size);
+                EXPECT_EQUAL(share_size, ecc_preferences->ecc_curves[i]->share_size);
 
                 EXPECT_SUCCESS(s2n_stuffer_skip_read(&key_share_extension, share_size));
             }
 
             EXPECT_SUCCESS(s2n_stuffer_free(&key_share_extension));
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
+        
+        /* Test s2n_extensions_client_key_share_send for a supported curve present in s2n_all_supported_curves_list,
+         * but not present in the ecc_preferences list selected */
+        {
+            if (s2n_is_evp_apis_supported()) {
+                struct s2n_stuffer key_share_extension;
+                struct s2n_connection *conn;
+                EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
+                EXPECT_NOT_NULL(conn->config);
+                /* Explicitly set the ecc_preferences list to contain the curves p-256 and p-384 */
+                EXPECT_SUCCESS(s2n_config_set_ecc_preferences(conn->config, "default"));
```

Comment:
So my last comment wasn't related to assuming that "default" is the default-- it's about assuming default doesn't contain x25519. "default" gives no guarantee of consistent behavior. We need to refer to the right preference list by its date name.